### ISSUE TITLE
[BUG] Upgrade Azure packages protobuf due to ContextualVersionConflict: protobuf 3.11.3 with Google API client

### ISSUE THREAD

**Issue by davidmytton** [Azure/azure-functions-python-worker#697]
### Investigative information
- Timestamp: `2020-06-21 12:43:44 UTC`
- Function App name: `dm-calgateway1-watch`
- Function name(s) (as appropriate): `watch`
- Core Tools version: `azure-functions-1.2.1`

### Repro steps
Include the following in `requirements.txt`

```
azure-functions
google-api-python-client==1.9.3
```

Then import the Google API client:

```
import logging

import azure.functions as func
import google.oauth2.credentials
import google_auth_oauthlib.flow

from googleapiclient.discovery import build
```

### Expected behavior
There should be no conflict.

### Actual behavior
The function fails with a fatal exception:

```
Time	13:43:44
Exception type	Microsoft.Azure.WebJobs.Host.FunctionInvocationException
Exception message	Exception while executing function: Functions.notification <--- Result: Failure Exception: ContextualVersionConflict: (protobuf 3.11.3 (/azure-functions-host/workers/python/3.7/LINUX/X64), Requirement.parse('protobuf>=3.12.0'), {'google-api-core'}) Stack: File "/azure-functions-host/workers/python/3.7/LINUX/X64/azure_functions_worker/dispatcher.py", line 259, in _handle__function_load_request func_request.metadata.entry_point) File "/azure-functions-host/workers/python/3.7/LINUX/X64/azure_functions_worker/utils/wrappers.py", line 29, in call return func(*args, **kwargs) File "/azure-functions-host/workers/python/3.7/LINUX/X64/azure_functions_worker/loader.py", line 73, in load_function mod = importlib.import_module(fullmodname) File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "<frozen importlib._bootstrap>", line 1006, in _gcd_import File "<frozen importlib._bootstrap>", line 983, in _find_and_load File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed File "<frozen importlib._bootstrap>", line 1006, in _gcd_import File "<frozen importlib._bootstrap>", line 983, in _find_and_load File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked File "<frozen importlib._bootstrap>", line 677, in _load_unlocked File "<frozen importlib._bootstrap_external>", line 728, in exec_module File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed File "/home/site/wwwroot/notification/__init__.py", line 8, in <module> from googleapiclient.discovery import build File "/home/site/wwwroot/.python_packages/lib/site-packages/googleapiclient/discovery.py", line 49, in <module> import google.api_core.client_options File "/home/site/wwwroot/.python_packages/lib/site-packages/google/api_core/__init__.py", line 23, in <module> __version__ = get_distribution("google-api-core").version File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 482, in get_distribution dist = get_provider(dist) File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 358, in get_provider return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0] File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 901, in require needed = self.resolve(parse_requirements(requirements)) File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 792, in resolve raise VersionConflict(dist, req).with_context(dependent_req)
ProcessId	29
prop__Succeeded	False
prop__TriggerReason	This function was programmatically called via the host APIs.
Microsoft.Azure.WebJobs.Host.FunctionInvocationException: Exception while executing function: Functions.notification
 ---> Microsoft.Azure.WebJobs.Script.Workers.Rpc.RpcException: Result: Failure
Exception: ContextualVersionConflict: (protobuf 3.11.3 (/azure-functions-host/workers/python/3.7/LINUX/X64), Requirement.parse('protobuf>=3.12.0'), {'google-api-core'})
Stack:   File "/azure-functions-host/workers/python/3.7/LINUX/X64/azure_functions_worker/dispatcher.py", line 259, in _handle__function_load_request
    func_request.metadata.entry_point)
  File "/azure-functions-host/workers/python/3.7/LINUX/X64/azure_functions_worker/utils/wrappers.py", line 29, in call
    return func(*args, **kwargs)
  File "/azure-functions-host/workers/python/3.7/LINUX/X64/azure_functions_worker/loader.py", line 73, in load_function
    mod = importlib.import_module(fullmodname)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/site/wwwroot/notification/__init__.py", line 8, in <module>
    from googleapiclient.discovery import build
  File "/home/site/wwwroot/.python_packages/lib/site-packages/googleapiclient/discovery.py", line 49, in <module>
    import google.api_core.client_options
  File "/home/site/wwwroot/.python_packages/lib/site-packages/google/api_core/__init__.py", line 23, in <module>
    __version__ = get_distribution("google-api-core").version
  File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 482, in get_distribution
    dist = get_provider(dist)
  File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 358, in get_provider
    return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
  File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)

   at Microsoft.Azure.WebJobs.Script.Description.WorkerFunctionInvoker.InvokeCore(Object[] parameters, FunctionInvocationContext context) in /src/azure-functions-host/src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs:line 85
   at Microsoft.Azure.WebJobs.Script.Description.FunctionInvokerBase.Invoke(Object[] parameters) in /src/azure-functions-host/src/WebJobs.Script/Description/FunctionInvokerBase.cs:line 85
   at Microsoft.Azure.WebJobs.Script.Description.FunctionGenerator.Coerce[T](Task`1 src) in /src/azure-functions-host/src/WebJobs.Script/Description/FunctionGenerator.cs:line 225
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`2.InvokeAsync(Object instance, Object[] arguments) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionInvoker.cs:line 52
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.InvokeAsync(IFunctionInvoker invoker, ParameterHelper parameterHelper, CancellationTokenSource timeoutTokenSource, CancellationTokenSource functionCancellationTokenSource, Boolean throwOnTimeout, TimeSpan timerInterval, IFunctionInstance instance) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 587
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithWatchersAsync(IFunctionInstanceEx instance, ParameterHelper parameterHelper, ILogger logger, CancellationTokenSource functionCancellationTokenSource) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 532
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance, ParameterHelper parameterHelper, IFunctionOutputDefinition outputDefinition, ILogger logger, CancellationTokenSource functionCancellationTokenSource) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 470
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance, FunctionStartedMessage message, FunctionInstanceLogEntry instanceLogEntry, ParameterHelper parameterHelper, ILogger logger, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 278
   --- End of inner exception stack trace ---
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance, FunctionStartedMessage message, FunctionInstanceLogEntry instanceLogEntry, ParameterHelper parameterHelper, ILogger logger, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 325
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.TryExecuteAsyncCore(IFunctionInstanceEx functionInstance, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 117
```

### Known workarounds
Switch to an older version of the Google API Python client prior to the latest protobuf release:

`google-api-python-client==1.7.11`

**Comment by mhadi813** [comment]
I'm also getting same error and the work around isn't working either:

this is my requirements.txt file
```
azure-functions==1.2.0                                            # tried latest version as well, same result
protobuf==3.11.3                                                      # tried 3.12.2, same result
google-api-python-client==1.7.11 
google-cloud-bigquery==1.24.0
grpcio==1.26.0
pandas==1.0.5

pyarrow==0.16.0
pytz==2019.3
pyodbc==4.0.27
requests==2.23.0

```

app log:

Result: Failure Exception: ContextualVersionConflict: (protobuf 3.11.3 (/azure-functions-host/workers/python/3.7/LINUX/X64), Requirement.parse('protobuf>=3.12.0'), {'google-api-core'}) Stack: File "/azure-functions-host/workers/python/3.7/LINUX/X64/azure_functions_worker/dispatcher.py", line 259, in _handle__function_load_request func_request.metadata.entry_point) File "/azure-functions-host/workers/python/3.7/LINUX/X64/azure_functions_worker/utils/wrappers.py", line 29, in call return func(*args, **kwargs) File "/azure-functions-host/workers/python/3.7/LINUX/X64/azure_functions_worker/loader.py", line 73, in load_function mod = importlib.import_module(fullmodname) File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "<frozen importlib._bootstrap>", line 1006, in _gcd_import File "<frozen importlib._bootstrap>", line 983, in _find_and_load File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed File "<frozen importlib._bootstrap>", line 1006, in _gcd_import File "<frozen importlib._bootstrap>", line 983, in _find_and_load File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked File "<frozen importlib._bootstrap>", line 677, in _load_unlocked File "<frozen importlib._bootstrap_external>", line 728, in exec_module File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed File "/home/site/wwwroot/mssql_reader/__init__.py", line 15, in <module> from .script import azure_to_bq File "/home/site/wwwroot/mssql_reader/script.py", line 4, in <module> from google.cloud import bigquery File "/home/site/wwwroot/.python_packages/lib/site-packages/google/cloud/bigquery/__init__.py", line 33, in <module> __version__ = get_distribution("google-cloud-bigquery").version File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 482, in get_distribution dist = get_provider(dist) File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 358, in get_provider return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0] File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 901, in require needed = self.resolve(parse_requirements(requirements)) File "/azure-functions-host/workers/python/3.7/LINUX/X64/pkg_resources/__init__.py", line 792, in resolve raise VersionConflict(dist, req).with_context(dependent_req)

**Comment by davidmytton** [comment]
This seems to be fixed with [the latest 1.1.4 release](https://github.com/Azure/azure-functions-python-worker/releases/tag/1.1.4) and `google-api-python-client 1.10.0`.

**Comment by Hazhzeng** [comment]
This issue has been fixed and rolled-out to all regions. Closing this issue but please let us know if you have face this again.
